src/content.rs

```diff
@@ -434,6 +422,63 @@ impl FromStr for Content {
     }
 }
 
+type ProcessFileReturn = (BTreeMap<String, PageValues>, Option<DateTime<Utc>>);
+
+fn process_files(
```

Comment:
`process` is a bit of an empty term when it comes to function names - it gives the reader no hint of _what kind of processing_ is being done. This can be okay when the function doesn't control what kind of processing is done (e.g. if the processing is passed in via a callback), but that's not the case here. Try a more specific name such as `read_file_contents`, `get_page_values`, `render_files` or something like that - you have a better idea than I do of what would be meaningful here!